packages/dd-trace/src/plugins/util/web.js

```diff
@@ -270,15 +270,26 @@ function wrapEnd (req) {
     return returnValue
   }
 
-  Object.defineProperty(res, 'end', {
-    configurable: true,
-    get () {
-      return this._datadog_end
-    },
-    set (value) {
-      this._datadog_end = scope.bind(value, req._datadog.span)
-    }
-  })
+  res._datadog_req = req
+  res._datadog_scope = scope
+
+  if (!res._datadog_instrumented) {
+    const target = Reflect.ownKeys(res).includes('end') ? res : res.constructor.prototype
```

Comment:
In which case would `res` have `end` directly if this is always expected to be the Node HTTP response object?

Reply:
In testing, I found that if I didn't do this, tests would fail. There are likely test cases somewhere in the suite that pass that kind of object in.

I can dig deeper to see if the test can be fixed.

Reply:
I dug deeper. Turns out `end` was always on the instance, due to being stubbed, so this was never being set on the prototype in tests. I fixed the test object so that `end`, even when stubbed, is always in a prototype instead of an instance.